src/HN/Model/Mailman.hs

```diff
@@ -0,0 +1,137 @@
+-- | construct a feed from a mailman (pipermail) archive,
+-- e.g., https://mail.haskell.org/pipermail/libraries/
+-- see https://github.com/chrisdone/haskellnews/issues/12
+-- and https://github.com/haskell-infra/hl/issues/63#issuecomment-74420850
+
+{-# language OverloadedStrings #-}
+{-# language BangPatterns #-}
+
+module HN.Model.Mailman where
+
+import Network.HTTP.Conduit
+-- I tried with
+-- import HN.Curl
+-- but could not get it to work reliably:
+-- segfaults, badcertfile, and other erratic behaviour
+-- on opening the second HTTPS connection.
+
+import Text.Feed.Types
+import Text.Feed.Constructor
+
+import qualified Text.HTML.DOM
+import Text.XML
+import Text.XML.Cursor
+import qualified Data.Text as T 
+
+import Data.Time.Format
+import Data.Time.LocalTime ( ZonedTime )
+import Data.Time ( rfc822DateFormat, defaultTimeLocale )
+import Data.Char (isDigit)
+
+import System.IO
+
+test1 = do
+  downloadFeed 10 "https://mail.haskell.org/pipermail/haskell-cafe/"
+
+
+-- | look at archive and produce a feed.
+-- 1st argument is minimal number of items to produce.
```

Comment:
Type synonyms don't help much - newtypes are better, but I don't care enough (in this case). HN.Model.Feeds.importGeneric uses String for URI as well.